src/map/map.ts

```diff
@@ -1140,9 +1140,23 @@ export class PropertiesMap {
             const defaultSize = this._is3D() ? 2000 : 150;
             values = sizes.map((v: number) => {
                 // normalize between 0 and 1, then scale by the user provided value
-                const scaled = userFactor * (v + 0.05 - min) / (max - min);
+                let scaled;
+                switch (scaleMode) {
```

Comment:
Instead of switching inside the `map` callback; if might be more efficient to use different callback for different modes:

```ts
switch (scaleMode) {
    case 'inverse':
        values = sizes.map((v: number) => (max - v + 0.05) / (max - min));
    case 'log':
        values = sizes.map((v: number) => Math.log((max - v + 0.05) / (max - min)));
...
}
```

I am not sure how much of a difference this will be in practice though

Reply:
I actually prefer how it's currently written so that you don't have to repeat `                return defaultSize * scaled * scaled * userFactor * userFactor;` for every single one.

Reply:
fine with me =)